### ISSUE TITLE
Flip camera while recording causes audio sync issues

### ISSUE THREAD

**Issue by AndresCanella** [NextLevel/NextLevel#16]
@piemonte I found the same issue I found in your [other lib](https://github.com/piemonte/PBJVision/issues/328) in this lib.

When you capture continuously and flip the camera mid capture. The audio get's chopped and looses sync to the video steam.

The ideal user experience would be for audio to keep recording without interruptions and get some black frames or a freeze frame while the camera is flipping(changing inputs). SnapChat keeps audio seamless so it's possible...

AVAudioRecorder, records audio uninterrupted even when flipping the camera. So it sounds like there might be a solution retiming the samples from the sample buffer?

I need to solve this for our lib in the next days so I'll keep you posted on our solution.

Any ideas?

**Comment by larspingtank** [comment]
@AndresCanella did you find a solution for this?

**Comment by anthonykim02** [comment]
flipping camera also results in bugs while merging video clips. is there any way to fix this?

**Comment by AndresCanella** [comment]
No, Have not found a solution that keeps audio intact. Currently I'm retiming the audio and video so that they are still in sync but the audio is missing a chunch when the camera flips. Will revisit this issue further down the line, will post here if I solve it before @piemonte. Perhaps using a separate audio recorder..

**Comment by jinksmysteryman** [comment]
@anthonykim02 I think I'm having some similar bugs with clip merging after flipping the camera. Did you find any solutions to this bug?

**Comment by anthonykim02** [comment]
@jinksmysteryman I wasn't able to fix it using Next Level. Instead I found a different library called SwiftyCam which didn't have merging video clips but did have hold to record. I had to implement the video clip merging myself, here's a link to SwiftyCam:
https://github.com/Awalz/SwiftyCam

**Comment by Kareemarab** [comment]
any solution for this issue so far?

**Comment by alezoffoli** [comment]
I also tried many things, like pausing/recording again before or after flipping and etc, but to no avail.

Any solution or at leads on how to fix?

**Comment by Woodyjl** [comment]
Hey all in my attempt to solve this issue in my own project I came up with a solution that works pretty well. Like @AndresCanella I am just retiming the audio frames. Hopefully this helps. [StackOverFlow response](https://stackoverflow.com/a/56572852/7090286)

**Comment by martino-dot** [comment]
Any implementation of the solution above this soon?

**Comment by NirajCapermint** [comment]
I am facing the issue of black screen for a second while flipping the camera. Any solution for this?
